talentsearch/resources/js/components/skills/SkillChecklist.tsx

```diff
@@ -0,0 +1,163 @@
+import React, { ChangeEvent } from "react";
+import groupBy from "lodash/groupBy";
+import { useIntl } from "react-intl";
+import { getLocale } from "@common/helpers/localize";
+import { SkillFamily } from "@common/api/generated";
+import {
+  getSkillCategoryText,
+  getSkillCategoryImage,
+} from "@common/constants/localizedConstants";
+
+interface FamilyProps {
+  family: SkillFamily;
+  index: string;
+  checked: boolean;
+  callback: (event: ChangeEvent<HTMLInputElement>) => void;
+}
+
+const Family: React.FunctionComponent<FamilyProps> = ({
+  family,
+  index,
+  checked,
+  callback,
+}) => {
+  const intl = useIntl();
+  const locale = getLocale(intl);
+
+  const uncheckedStyle = "data-h2-font-weight='b(400)'";
+  const checkedStyle = "data-h2-font-weight='b(700)'";
+
+  return (
+    <div
+      data-h2-font-weight={checked ? checkedStyle : uncheckedStyle}
+      data-h2-padding="b(all, xxs)"
+      key={family.key}
+    >
+      <input
+        type="checkbox"
+        name={index}
+        id={family.key}
+        onChange={(e) => {
+          callback(e);
+        }}
+      />
+      &nbsp;
+      <label htmlFor={family.key}>
+        {family.name?.[locale]} ({family.skills ? family.skills.length : 0})
+      </label>
+    </div>
+  );
+};
+
+interface CategoryProps {
+  category: string;
+  skillFamilies: SkillFamily[];
+  callback: (selected: SkillFamily, added: boolean) => void;
+}
+
+const Category: React.FunctionComponent<CategoryProps> = ({
+  category,
+  skillFamilies,
+  callback,
+}) => {
+  const intl = useIntl();
+
+  // The app currently crashes if no description is specified for the category.
+  // Do we want to avoid this?
+  let title = intl.formatMessage(getSkillCategoryText(category));
+
+  const image = getSkillCategoryImage(category);
+
+  // Calculate the total number of skills in the SkillCategory
+  let total = 0;
+  skillFamilies.forEach((family) => {
+    total += family.skills ? family.skills.length : 0;
+  });
+  title += ` (${total})`;
+
+  const [checkedArr, setCheckedArr] = React.useState<string[]>([]);
+
+  const HandleChecked = (event: ChangeEvent<HTMLInputElement>) => {
+    const pos = Number(event.target.name);
+    if (event.target.checked) {
+      setCheckedArr([...checkedArr, skillFamilies[pos].key]);
```

Comment:
Good use of [spread syntax](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax)!

Reply:
Thanks!